### PR TITLE
Keshav week4

### PR DESCRIPTION
**Pull request by choudhary-keshav**
Pull Request Questionnaire

(1) Which Jira ticket(s) is this pull request for, and is the ticket now complete?

(2) Briefly describe your change.
I have integrated my week-4 assignment in this joint project. In my week-4 assignment I have added a weather component for knowing whether details of any city by OpenWeatherMap API, and also made a Tic-Tac-Toe game and added dynamic routing for navigating between these two components.

(3) What parts of the application are affected by your change?
It'd just affect the "Keshav's", "week-4" component.

(4) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Tested my application on various edge cases and some common scenarios

Root Cause (if any)?
-

### CODE REVIEW

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/TicTacToe/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    if (checkWinner(newBoard, player === 1 ? 'X' : 'O')) {
+      setWinner(player);
+      return;
+    }
+
+    setPlayer(player === 1 ? 2 : 1);
+    setTicTacToeBoard(newBoard);
+  };
+
+  const checkWinner = (board: (number | string)[][], symbol: string): boolean => {
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/TicTacToe/index.tsx]: since symbol is a JS datatype I would suggest choosing a different name for this argument to avoid confusion. I think you dont need that argument
you can do something like 
if (board[i][0] === board[i][1]  && board[i][1] === board[i][2]) return board[i][0]
but here you might have to handle the empty case

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/TicTacToe/styles.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        font-size: 60px;
+        cursor: pointer;
+    }
+
+    button{
+        height: 100px;
+        width: 100px;
+    }
+
+`
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/TicTacToe/styles.ts:28]: Please add eol, add it in your eslintrc file if you dont already have it

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  });
+
+  const apiKey = process.env.REACT_APP_WEATHER_API_KEY;
+  const url = 'https://api.openweathermap.org/data/2.5/weather?q=' + cityName + '&appid=' + apiKey + '&units=metric';
+
+  const fetchTemperature = () => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx:25]: remove consoles

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        setApiCode(data.cod);
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx:26]: you can convert your statusCode to number before saving to have apiCode as only number value like this
const [apiCode, setApiCode] = useState<number>(-1);

File: interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          <p>Latitude: {weatherDetails.latitude}</p>
+          <p>Longitude: {weatherDetails.longitude}</p>
+          <p>Country: {weatherDetails.country}</p>
+        </div>
+      )}
+
+      {apiCode &&
+        cityName &&
+        (apiCode === '404' ? (
+          <p className='apiError'>City not found!</p>
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx:74]: You can start using a label/constants file to save these messages

**Comment by anusreesubash** [comment]
Also add package-lock.json to gitignore file
